services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceControllerClusterConfig.java

```diff
@@ -894,7 +905,33 @@ public VeniceControllerClusterConfig(VeniceProperties props) {
     this.emergencySourceRegion = props.getString(EMERGENCY_SOURCE_REGION, "");
     this.allowClusterWipe = props.getBoolean(ALLOW_CLUSTER_WIPE, false);
     this.concurrentInitRoutinesEnabled = props.getBoolean(CONCURRENT_INIT_ROUTINES_ENABLED, false);
-    this.controllerInAzureFabric = props.getBoolean(CONTROLLER_IN_AZURE_FABRIC, false);
+    this.controllerClusterHelixCloudEnabled = props.getBoolean(CONTROLLER_CLUSTER_HELIX_CLOUD_ENABLED, false);
+    this.controllerStorageClusterHelixCloudEnabled =
+        props.getBoolean(CONTROLLER_STORAGE_CLUSTER_HELIX_CLOUD_ENABLED, false);
+
+    if (controllerClusterHelixCloudEnabled || controllerStorageClusterHelixCloudEnabled) {
+      String controllerCloudProvider = props.getString(CONTROLLER_HELIX_CLOUD_PROVIDER, "").toUpperCase();
```

Comment:
Since we need this field to be set if `controllerClusterHelixCloudEnabled` or `controllerStorageClusterHelixCloudEnabled`, can we remove the default value of `""` ? That way we ensure that this is always provided